// dustjs-brunch/lib/index.js
(function() {
  var DustCompiler, dust, sysPath;

  dust = require('dust-formspring');

  sysPath = require('path');

  module.exports = DustCompiler = (function() {

    DustCompiler.prototype.brunchPlugin = true;

    DustCompiler.prototype.type = 'template';

    DustCompiler.prototype.extension = 'dust';

    function DustCompiler(config) {
      this.config = config;
      null;
    }

    DustCompiler.prototype.compile = function(data, path, callback) {
      var content, content_string, error, file_name, result, split_path;
      try {
        split_path = path.split('templates/');
        if (split_path.length > 1) {
          file_name = split_path[split_path.length - 1];
        } else {
          file_name = split_path[0];
        }
        content = dust.compile(data, file_name.replace(/\.dust/, ''));
        content_string = JSON.stringify(content);
        return result = "module.exports = " + content_string + ";";
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    DustCompiler.prototype.include = function() {
      return [sysPath.join(__dirname, '..', 'node_modules', 'dust-formspring', 'lib', 'dust.js')];
    };

    return DustCompiler;

  })();

}).call(this);
